diff --git a/dist/index.d.ts b/dist/index.d.ts
index bfd78356999ab49ee98320bb7bf759eb9f55fd12..da4596431570bcdd000eadc3fb1c5617c06c5d27 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,4 +1,5 @@
 /// <reference types="react" />
+import { Stable } from '@animedes/react-stable';
 import * as motion_dom from 'motion-dom';
 import { AnyResolvedKeyframe, OnKeyframesResolved, KeyframeResolver, MotionNodeOptions, MotionValue, TransformProperties, SVGPathProperties, Transition, PresenceContextProps, ResolvedValues, VisualElement, Feature, UnresolvedValueKeyframe, AnimationOptions, ElementOrSelector, DOMKeyframesDefinition, AnimationPlaybackOptions, AnimationPlaybackControlsWithThen, ValueAnimationTransition, addEffect, removeEffect, AnimationScope, AnimationPlaybackControls, EventInfo, MotionValueEventCallbacks, FollowValueOptions, SpringOptions, TransformOptions, WillChange, LegacyAnimationControls, NodeGroup, IProjectionNode } from 'motion-dom';
 export * from 'motion-dom';
@@ -1053,7 +1054,7 @@ declare function useMotionTemplate(fragments: TemplateStringsArray, ...values: A
  *
  * @public
  */
-declare function useMotionValue<T>(initial: T): MotionValue<T>;
+declare function useMotionValue<T>(initial: T): Stable<MotionValue<T>>;
 
 interface UseScrollOptions extends Omit<ScrollInfoOptions, "container" | "target"> {
     container?: RefObject<HTMLElement | null>;
