diff --git a/wrangler-dist/cli.js b/wrangler-dist/cli.js
index a6c9d186014bb4868a177b0fda3712f52cd0bbc0..002ff9eb4a74b243ebf09bdef44881e52ff6eaf1 100644
--- a/wrangler-dist/cli.js
+++ b/wrangler-dist/cli.js
@@ -29569,15 +29569,6 @@ var require_wasm_sync = __commonJS({
           }, "ha");
           1 < process.argv.length && process.argv[1].replace(/\\/g, "/");
           process.argv.slice(2);
-          process.on("uncaughtException", function(a) {
-            throw a;
-          });
-          process.on(
-            "unhandledRejection",
-            function(a) {
-              throw a;
-            }
-          );
           l2.inspect = function() {
             return "[Emscripten Module object]";
           };
